### PR TITLE
fix(server): Reduce TTL for negative DNS cache to 3 seconds

### PR DESCRIPTION
**Pull request by jan-auer**

### CODE REVIEW

**Comment by jan-auer** [comment]
The idea of this change was to speculatively fix errors in onpremise where Relay is not able to resolve the web container. However, after trying to reproduce this with the latest `master`, I am no longer sure this is the correct fix. At least, I am not able to reproduce the bug in our docker-compose setup when I manually start the `web` container late -- Relay simply connects with the next authentication retry.

File: relay-config/src/config.rs [hunk 1]
```diff
@@ -550,6 +579,7 @@ impl Default for Http {
             host_header: None,
             auth_interval: Some(600), // 10 minutes
             auth_grace_period: 10,
+            encoding: HttpEncoding::Gzip,
```
Comment (BYK) [inline relay-config/src/config.rs]: Is this PR sneaking in an unrelated change (adding GZip support for upstream request body) or is it me?

**Review by BYK — APPROVED** [review]
Wow, this is a lot of code for reducing a TTL :D

File: relay-config/src/config.rs (hunk 1 above)
Comment (jan-auer) [reply relay-config/src/config.rs]: You're right, pushed a commit to the wrong branch.

File: relay-server/src/actors/connector.rs [hunk 2]
```diff
@@ -43,6 +46,24 @@ impl MeteredConnector {
...
+        //     fixed by the author.
+        //  2. A service like load balancer or docker container starting up with delay. In such a
+        //     case, Relay is expected to connect as soon as the service becomes available.
+        //  3. Misconfiguration in the network, most likely DNS server.
+        //
+        // By default, Relay detects these issues during startup time and employs exponential
+        // backoff to reconnect. Since outgoing traffic is of low volume at this point, the TTL for
+        // negative DNS caches can be set to a few seconds.
+        let mut opts = ResolverOpts::default();
+        opts.negative_max_ttl = Some(Duration::from_secs(3));
```
Comment (tonyo) [inline relay-server/src/actors/connector.rs:62]: What's the default value?

Comment (jan-auer) [reply relay-server/src/actors/connector.rs:62]: I'm not 100% sure, but the default value seems to be 1 day.

**Comment by jan-auer** [comment]
I was not able to reproduce the underlying cause. Until we can reproduce and correlate, I am no longer planning to merge this PR.
